samples/Azure.ResourceManager.Sample/Generated/Extensions/SubscriptionResourceExtensionClient.cs

```diff
@@ -490,24 +490,18 @@ public virtual Response<VirtualMachineImage> GetVirtualMachineImage(AzureLocatio
         /// Request Path: /subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/publishers/{publisherName}/artifacttypes/vmimage/offers/{offer}/skus/{skus}/versions
         /// Operation Id: VirtualMachineImages_List
         /// </summary>
-        /// <param name="location"> The name of a supported Azure region. </param>
-        /// <param name="publisherName"> A valid image publisher. </param>
-        /// <param name="offer"> A valid image publisher offer. </param>
-        /// <param name="skus"> A valid image SKU. </param>
-        /// <param name="expand"> The expand expression to apply on the operation. </param>
-        /// <param name="top"> The Integer to use. </param>
-        /// <param name="orderby"> The String to use. </param>
+        /// <param name="options"> A property bag which contains all the parameters of this method except the LRO qualifier and request context parameter. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <returns> An async collection of <see cref="VirtualMachineImageResource" /> that may take multiple service requests to iterate over. </returns>
-        public virtual AsyncPageable<VirtualMachineImageResource> GetVirtualMachineImagesAsync(AzureLocation location, string publisherName, string offer, string skus, string expand = null, int? top = null, string orderby = null, CancellationToken cancellationToken = default)
+        public virtual AsyncPageable<VirtualMachineImageResource> GetVirtualMachineImagesAsync(SampleExtensionsGetVirtualMachineImagesOptions options, CancellationToken cancellationToken = default)
```

Comment:
It appears in other classes like [here](https://github.com/Azure/autorest.csharp/pull/2862/files#diff-f5f2f87eae78ca10a21df07fa50fcb686542ca8652442cece0ee4fe39ffed17dR197) we are defaulting the options bag if its null since everything is optional, and then [here](https://github.com/Azure/autorest.csharp/pull/2862/files#diff-5bb05231d693a0ca35041f241a2ad7978d20ae6610dcae4f8a08d4930ddb211cR249) we are asserting not null since some of the parameters are required.

Why are we not doing either of those for this method?

Reply:
Actually, we did, this asserting happened in the [public extension](https://github.com/Azure/autorest.csharp/blob/d76abeb8c49264eef9de1c8c6dd05f01b9c25f2b/samples/Azure.ResourceManager.Sample/Generated/Extensions/SampleExtensions.cs#L249).

Reply:
Ah didn't realize I was looking at the internal extension client and the validation goes on the public interface.